File: duck.py
from selenium import webdriver
import os

chrome_options = webdriver.ChromeOptions()
chrome_options.binary_location = os.environ.get("GOOGLE_CHROME_BIN")
chrome_options.add_argument("--headless")
chrome_options.add_argument("--disable-dev-shm-usage")
chrome_options.add_argument("--no-sandbox")
driver = webdriver.Chrome(executable_path=os.environ.get("CHROMEDRIVER_PATH"), chrome_options=chrome_options)


driver.get("http://www.duckduckgo.com/")
print (driver.page_source)
#search_box = driver.get_element_by_id("search_form_input_homepage")
#search_box.driver.clear()
#search_box.send_keys("frz_akbar instagram")

#print (search_box)
#search_box.send_keys(Keys.RETURN)
#submit = driver.get_element_by_id("search_button_homepage")
#submit.click()

File: duck.js
const webdriver = require('selenium-webdriver');
// const chrome = require(selenium-webdriver/chrome);
console.log("masuk ke function")
(async function getduck() {
  // let driver = await new Builder().forBrowser('chrome').build();
  let chromeOptions = await new webdriver.setChromeOptions()
    .setChromeBinaryPath('/app/.apt/usr/bin/google-chrome')
    .headless()
    .addArguments('disable-dev-shm-usage')
    .addArguments('no-sandbox');
    console.log("chrome option done")
  
  let driver = await new webdriver.Builder()
    .forBrowser('chrome')
    .execute('/app/.chromedriver/bin/chromedriver')
    .setChromeOptions(chromeOptions)
    .build();
    console.log("build chrome done")
  try {
    // Navigate to Url
    await driver.get('https://duckduckgo.com');
        console.log("loading done");
        // Get all the elements available with tag name 'p'
        await driver.findElement(By.id('search_form_input_homepage')).sendKeys('Frz_akbar Instagram', Key.ENTER);
        // for(let e of elements) {
        //     console.log(await e.getText());
        // }
    }
    finally {
        console.log("done");
        // await driver.quit();
    }
})();
  


